app.rb

```diff
@@ -0,0 +1,160 @@
+include Math
+
+# addition function
+def addition (num1, num2)
+  puts num1 + num2
+end
+
+# subtraction function
+def subtraction (num1, num2)
+  puts num1 - num2
+end
+
+# multiplication function
+def multiplication (num1, num2)
+  puts num1 * num2
+end
+
+# division function
+def division (num1, num2)
+  puts num1 / num2
+end
+
+# power function
+def power (num1, num2)
+  puts num1**num2
+end
+
+# square root function
+def square_root (num1)
+  puts sqrt(num1)
+end
+
+# Imperial BMI function
+def imperial (weight, height)
+  puts 703*weight/(height**2)
+end
+
+# metric BMI function
+def metric (weight, height)
+  puts weight/(height**2)
+end
+
+# time trip function
+def time (dist, speed)
+  dist / speed
+end
+
+# final cost function
+def final_cost (speed, mpg, cost)
+  if speed > 60 && mpg > speed && mpg > 0
+    mpg = mpg - ((speed - 60) * 2)
+    return mpg * cost
+  else mpg * cost
+  end
+end
+
+# start of calculator
+puts "Welcome, would you like to the advanced, basic calculator, BMI or trip"
```

Comment:
These options are quite obtuse - it's a bit much to expect the user to perfect type out full words like "advanced" and "addition". I would make them options like (a) and (b) or (1) and (2).